### PR TITLE
Added tests for StatusMetadataExtractor.

### PR DESCRIPTION
**Pull request by fiboknacky**
The content for all versions is the same.

### CODE REVIEW

File: tests/Google/Ads/GoogleAds/Lib/V1/StatusMetadataExtractorTest.php [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+     * @see \PHPUnit\Framework\TestCase::setUp()
+     */
+    protected function setUp()
+    {
+        $this->statusMetadataExtractor = new StatusMetadataExtractor();
+    }
+
+    public function testExtractGoogleAdsFailure()
+    {
+        $expected = new GoogleAdsFailure();
```
Comment (PierrickVoulet) [inline tests/Google/Ads/GoogleAds/Lib/V1/StatusMetadataExtractorTest.php:49]: The creation of this object is done at least two times in the tests (`testExtractGoogleAdsFailure ` and `testExtractErrorMessageList `), could you factorize?

Ditto in V2 and V3.

**Review by PierrickVoulet — CHANGES_REQUESTED** [review]
We got many issues reported about the extraction of errors. Do these tests cover it all?

Also, IIRC there is a big difference in format whether REST or gRPC is used, do we want to add tests specifically for REST here?

**Comment by fiboknacky** [comment]
> We got many issues reported about the extraction of errors. Do these tests cover it all?
> 
> Also, IIRC there is a big difference in format whether REST or gRPC is used, do we want to add tests specifically for REST here?

No. It's not an aim of this PR. This PR is just for adding tests of existing classes.

**Comment by PierrickVoulet** [comment]
> > We got many issues reported about the extraction of errors. Do these tests cover it all?
> > Also, IIRC there is a big difference in format whether REST or gRPC is used, do we want to add tests specifically for REST here?
> 
> No. It's not an aim of this PR. This PR is just for adding tests of existing classes.

Thanks for confirming, we are good then. I let you open issue(s) if you wish so that we can prioritize what is not in the scope of this PR later.

**Review by PierrickVoulet — APPROVED** [review]

**Comment by fiboknacky** [comment]
> > > We got many issues reported about the extraction of errors. Do these tests cover it all?
> > > Also, IIRC there is a big difference in format whether REST or gRPC is used, do we want to add tests specifically for REST here?
> > 
> > 
> > No. It's not an aim of this PR. This PR is just for adding tests of existing classes.
> 
> Thanks for confirming, we are good then. I let you open issue(s) if you wish so that we can prioritize what is not in the scope of this PR later.

In fact, I quickly check the code again and it seems everything is already in place. It allows for extracting both grpc and REST keys.
Based on the previous discussions, I found most complaints were usually based on misunderstanding in some ways.

**Pull request merged**